python/podio/frame.py

```diff
@@ -10,49 +10,52 @@
-          'c++' or 'py'.
-
   Returns:
-      tuple (str): the tuple with the string representation of all **c++**
-          classes that are supported
+      tuple(tuple(str, str)): the tuple with the string representation of all
+          c++ and their corresponding python types that are supported
   """
   types_tuple = podio.SupportedGenericDataTypes()
   n_types = cppyy.gbl.std.tuple_size[podio.SupportedGenericDataTypes].value
 
   # Get the python types with the help of cppyy and the STL
-  py_types = (type(cppyy.gbl.std.get[i](types_tuple)).__name__ for i in range(n_types))
-  if lang == 'py':
-    return tuple(py_types)
-  if lang == 'c++':
-    # Map of types that need special care when going from python to c++
-    py_to_cpp_type_map = {
-        'str': 'std::string'
-        }
-    # Convert them to the corresponding c++ types
-    return tuple(py_to_cpp_type_map.get(t, t) for t in py_types)
+  py_types = [type(cppyy.gbl.std.get[i](types_tuple)).__name__ for i in range(n_types)]
+
+  def _determine_cpp_type(idx_and_type):
+    """Determine the actual c++ type from the python type name.
+
+    Mainly maps 'str' to 'std::string', and also determines whether a python
+    'float' is actually a 'double' or a 'float' in c++. The latter is necessary
+    since python only has float (corresponding to double in c++) and we
+    need the exact c++ type
+    """
+    idx, typename = idx_and_type
+    if typename == 'float':
+      cpp_type = cppyy.gbl.std.tuple_element[idx, podio.SupportedGenericDataTypes].type
+      if cppyy.typeid(cpp_type).name() == 'd':
+        return 'double'
+      return 'float'
+    if typename == 'str':
+      return 'std::string'
+    return typename
 
-  raise ValueError(f"lang needs to be 'py' or 'c++' (got {lang})")
+  cpp_types = list(map(_determine_cpp_type, enumerate(py_types)))
+  return tuple(zip(cpp_types, py_types))
 
 
-SUPPORTED_PARAMETER_TYPES = _determine_supported_parameter_types('c++')
-SUPPORTED_PARAMETER_PY_TYPES = _determine_supported_parameter_types('py')
+SUPPORTED_PARAMETER_TYPES = _determine_supported_parameter_types()
 
 
-# Map that is necessary for easier disambiguation of parameters that are
-# available with more than one type under the same name. Maps a python type to
-# a c++ vector of the corresponding type or a c++ type to the vector
-_PY_TO_CPP_TYPE_MAP = {
-    pytype: f'std::vector<{cpptype}>' for (pytype, cpptype) in zip(SUPPORTED_PARAMETER_PY_TYPES,
-                                                                     SUPPORTED_PARAMETER_TYPES)
-    }
-_PY_TO_CPP_TYPE_MAP.update({
-    f'{cpptype}': f'std::vector<{cpptype}>' for cpptype in SUPPORTED_PARAMETER_TYPES
-    })
+def _get_cpp_vector_types(type_str):
+  """Get the possible std::vector<cpp_type> from the passed py_type string."""
+  # Gather a list of all types that match the type_str (c++ or python)
+  types = list(filter(lambda t: type_str in t, SUPPORTED_PARAMETER_TYPES))
```

Comment:
```suggestion
  types = [t for t in SUPPORTED_PARAMETER_TYPES if type_str in t]
```
More pythonic and I find list comprehensions more readable than compositions of `list(map(filter`...